addons/sale_pdf_quote_builder/models/sale_order.py

```diff
@@ -48,6 +49,19 @@ def _compute_is_pdf_quote_builder_available(self):
                 or order.order_line.available_product_document_ids
             )
 
+    # === ONCHANGE METHODS === #
+
+    @api.onchange('sale_order_template_id')
+    def _onchange_sale_order_template_id(self):
+        super()._onchange_sale_order_template_id()
+        for order in self:
+            available_quotation_documents = self.env['quotation.document'].search(AND([
+                self.env['quotation.document']._check_company_domain(order.company_id),
+                [('quotation_template_ids', 'in', [False, order.sale_order_template_id.id])]
+            ]))
+            # Remove documents which are no longer available.
+            self.quotation_document_ids &= available_quotation_documents
```

Comment:
```suggestion
        available_quotation_documents = self.env['quotation.document'].search(AND([
            self.env['quotation.document']._check_company_domain(order.company_id),
            [('quotation_template_ids', 'in', [False, self.sale_order_template_id.id])]
        ]))
        # Remove documents which are no longer available.
        self.quotation_document_ids &= available_quotation_documents
```
It's mixing up `self` and `order`. I think `onchange` methods aren't expected to handle multiple records, so you can just remove the `for`, otherwise change the `self` in the last line to `order`

Reply:
It's a common non-written agreement to either manage in batch OR have the `self.ensure_one()` OR a for loop, even when we know it's a single record. For this one, I'd let the for loop. =]

Also, using `self` and `order` is fine. It shows when it's something not specific to that specific order, and when it is. So I'd let as it was except for that last `self` that should be `order` instead